packages/flutter/lib/src/widgets/actions.dart

```diff
@@ -1153,3 +1153,22 @@ class SelectIntent extends Intent {}
 /// This is an abstract class that serves as a base class for actions that
 /// select something. It is not bound to any key by default.
 abstract class SelectAction extends Action<SelectIntent> {}
+
+/// An [Intent] that cancels or dismisses the currently focused widget.
+///
+/// By default, this intent is bound to [LogicalKeyboardKey.escape],
+/// and [LogicalKeyboardKey.gameButtonB] in the
+/// default keyboard shortcuts in [WidgetsApp].
+///
+/// See also:
+///  - [ModalRoute] which listens for this intent to dismiss modal routes
+///    (dialogs, pop-up menus, drawers, etc).
+class CancelIntent extends Intent {
+  /// Creates a const [CancelIntent].
+  const CancelIntent();
+}
+
+/// An action that cancels or dismisses the focused widget.
+///
+/// This is an abstract class that serves as a base class for cancel actions.
+abstract class CancelAction extends Action<CancelIntent> {}
```

Comment:
Why call this cancel instead of dismiss?

Reply:
I actually started with dismiss, but it seems like it could be more generic as a cancel/negative action/dismiss feature. It is fairly common in game interfaces to have the A button mapped to 'activate or select the focused thing', and the B button mapped to 'cancel/go back/dismiss'. That was my thinking, but if others feel strongly it should just be dismiss, I can put it  back.

Reply:
I don't have a strong opinion about it; just noticed that in most of your descriptions of this intent/action you used "dismiss".

Reply:
I think dismiss is better.  Cancel and dismiss are slightly different things, and this is definitely a dismiss.  Cancel implies that the user specifically doesn't want something to happen, or wants to interrupt something in progress. Dismiss means that they just want to ignore something or to acknowledge that they saw something and put it away. Users can always create a separate cancel action for the other sense if they need it.

Reply:
Sounds good. I just renamed them back to Dismiss*. Let me know what you think.